src/os/bluestore/bluestore_types.h

```diff
@@ -283,7 +283,7 @@ struct bluestore_blob_t {
     }
     b_len += b_off;
     while (b_len) {
-      if (!p->is_valid()) {
+      if (p == extents.end() || !p->is_valid()) {
```

Comment:
we should probably remove assert above (line 282) and return false too when iterator reaches the end.